key_file12.txt

```diff
@@ -0,0 +1,50 @@
+--- Test 12 output ---
+
+Pretty printing the tree...
+Empty tree
+Inserting 6, 4, 7, 2, 5, 9, 1, and 3 into the tree...
+Pretty printing the tree...
+
+               6
+
+       4               7
+
+   2       5               9
+
+ 1   3                        
+
+Attempting to remove 9 from the tree...true
+Pretty printing the tree...
+
+       4
+
+   2       6
+
+ 1   3   5   7
+
+Checking that 4 is at the head of the tree...true
+Attempting to remove 4 from the tree...true
+
+Checking that 3 is at the head of the tree...true
+Attempting to remove 3 from the tree...true
+
+Checking that 2 is at the head of the tree...true
+Attempting to remove 2 from the tree...true
+
+Attempting to remove 4 from the tree...false
+
+Checking that 6 is at the head of the tree...true
+Attempting to remove 6 from the tree...true
+
+Checking that 5 is at the head of the tree...true
+Attempting to remove 5 from the tree...true
+
+Checking that 1 is at the head of the tree...true
+Attempting to remove 1 from the tree...true
+
+Checking that 7 is at the head of the tree...true
+Attempting to remove 7 from the tree...true
+
+Pretty printing the tree...
+
+Empty tree
```

Comment:
Let's have a separate test like this and remove this part (same for other similar tests). Most tests should focus on a single unit of functionality, with maybe 1 or 2 tests that focus on comprehensive performance.